src/Dfc.CourseDirectory.FindACourseApi/Features/TLevels/TLevels.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.Sql;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using MediatR;
+using CosmosQueries = Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+
+namespace Dfc.CourseDirectory.FindACourseApi.Features.TLevels
+{
+    public class Query : IRequest<ViewModel>
+    {
+    }
+
+    public class ViewModel
+    {
+        public IReadOnlyCollection<TLevelDetailViewModel> TLevels { get; set; }
+    }
+
+    public class Handler : IRequestHandler<Query, ViewModel>
+    {
+        private readonly ISqlQueryDispatcher _sqlQueryDispatcher;
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+
+        public Handler(ISqlQueryDispatcher sqlQueryDispatcher, ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        {
+            _sqlQueryDispatcher = sqlQueryDispatcher;
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+        }
+
+        public async Task<ViewModel> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var tLevels = await _sqlQueryDispatcher.ExecuteQuery(new GetTLevels());
```

Comment:
I'm nervous about the performance of this approach once we begin to see large number of T Levels available within the system.

Reply:
Agreed - we could look into the `IAsyncEnumerable` from DB -> JSON in future but for a first pass this seems fine.